src/Tests/AutologoutTestCaseTest.php

```diff
@@ -0,0 +1,228 @@
+  // Check that the user can access the page after login.
+  $this->drupalGet('node');
+  $this->assertResponse(200, t('Homepage is accessible'));
+  $this->assertText(t('Log out'), t('User is still logged in.'));
+
+  // Wait within the timeout period.
+  sleep(10);
+
+  // Check we are still logged in.
+  $this->drupalGet('node');
+  $this->assertResponse(200, t('Homepage is accessible'));
+  $this->assertText(t('Log out'), t('User is still logged in.'));
+  $this->assertNoText(t('You have been logged out due to inactivity.'), t('User does not see inactivity message.'));
+  }
+
+  /**
+  * Test the behaviour of the settings for submission.
+  */
+  public function testAutologoutSettingsForm() {
+    $edit = array();
+    $autologout_settings = \Drupal::config('autologout.settings');
+    $autologout_settings->set('max_timeout', 1000)
+      ->save();
+
+  // Test that it is possible to set a value above the max_timeout
+  // threshold.
+  $edit['timeout'] = 1500;
+  $edit['max_timeout'] = 2000;
+  $edit['padding'] = 60;
+  $edit['role_logout'] = TRUE;
+  $edit['autologout_redirect_url'] = TRUE;
+
+  $this->drupalPostForm('admin/config/people/autologout', $edit, t('Save configuration'));
+  $this->assertText(t('The configuration options have been saved.'), t('Unable to save autologout config when modifying the max timeout.'));
+
+  // Test that out of range values are picked up.
+  $edit['autologout_timeout'] = 2500;
+  $edit['autologout_max_timeout'] = 2000;
+  $edit['autologout_padding'] = 60;
+  $edit['autologout_redirect_url'] = TRUE;
+
+  $this->drupalPostForm('admin/config/people/autologout', $edit, t('Save configuration'));
+  $this->assertNoText(t('The configuration options have been saved.'), t('Saved configuration despite the autologout_timeout being too large.'));
+
+  // Test that out of range values are picked up.
+  $edit['autologout_timeout'] = 1500;
+  $edit['autologout_max_timeout'] = 2000;
+  $edit['autologout_padding'] = 60;
+  $edit['autologout_redirect_url'] = TRUE;
+
+  $this->drupalPostForm('admin/config/people/autologout', $edit, t('Save configuration'));
+  $this->assertNoText(t('The configuration options have been saved.'), t('Saved configuration despite a role timeout being too large.'));
+
+  // Test that role timeouts are not validated for
+  // disabled roles.
+  $edit['autologout_timeout'] = 1500;
+  $edit['autologout_max_timeout'] = 2000;
+  $edit['autologout_padding'] = 60;
+  $edit['autologout_role_logout'] = TRUE;
+  $edit['autologout_redirect_url'] = TRUE;
+
+  $this->drupalPostForm('admin/config/people/autologout', $edit, t('Save configuration'));
+  $this->assertText(t('The configuration options have been saved.'), t('Unable to save autologout due to out of range role timeout for a role which is not enabled..'));
+  }
+
+  /**
+  * Test enforce logout on admin page settings.
+  */
+  public function testAutlogoutOfAdminPages() {
+
+  // Set an admin page path.
+  $_GET['q'] = 'admin';
```

Comment:
Indent the code by 4 spaces here. Also check throughout the file for the indentation.